frontstage/views/sign_in.py

```diff
@@ -29,27 +29,49 @@ def login():
         username = request.form.get('username')
         password = request.form.get('password')
 
-        # Creates a 'session client' to interact with OAuth2. This provides a client ID to our client that is used to
-        # interact with the server.
-        client = LegacyApplicationClient(client_id=app.config['RAS_FRONTSTAGE_CLIENT_ID'])
-
-        # Populates the request body with username and password from the user
-        client.prepare_request_body(username=username, password=password, scope=['ci.write', 'ci.read'])
-
-        # passes our 'client' to the session management object. this deals with
-        # the transactions between the OAuth2 server
-        oauth = OAuth2Session(client=client)
-        token_url = app.config['ONS_OAUTH_PROTOCOL'] + app.config['ONS_OAUTH_SERVER'] + app.config['ONS_TOKEN_ENDPOINT']
-
+        # TODO Consider moving this to a helper function.
+        # Lets get a token from the OAuth2 server
         try:
-            token = oauth.fetch_token(token_url=token_url, username=username, password=password, client_id=app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                                      client_secret=app.config['RAS_FRONTSTAGE_CLIENT_SECRET'])
-
+            token_url = app.config['ONS_OAUTH_PROTOCOL'] + app.config['ONS_OAUTH_SERVER'] + app.config['ONS_TOKEN_ENDPOINT']
+            data = {
+                'grant_type': 'password',
+                'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
+                'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
+                'username': username,
+                'password': password,
+            }
+            headers ={
+                'Accept': 'application/json',
+                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+            }
+
+            oauth2_response = requests.post(url=token_url, data=data , headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'], app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
+            oauth2_token = json.loads(oauth2_response.text)
+            # Check to see that this user has not attempted to login too many times or that they have not forgot to
+            # click on the activate account URL in their email by checking the error message back from the OAuth2 server
+            if oauth2_response.status_code == 401:
+                oauth2Error = json.loads(oauth2_response.text)
+                if oauth2Error['detail']== 'Unauthorized user credentials':
+                    return render_template('sign-in/sign-in.html', _theme='default', form=form, data={"error": {"type": "failed"}})
+                elif 'User account locked' in oauth2Error['detail']:
+                    logger.warning('User account is locked on the OAuth2 server')
+                    return render_template('sign-in/sign-in.trouble.html', _theme='default', form=form,
```

Comment:
It's the one specified by Nick UI. He presented a specific type of text bla bla bla...